elisa/src/client/RunDetails/components/CalibrationCurvePanel.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { PureComponent } from 'react';
+import $ from 'jquery';
+import { getServerContext } from '@labkey/api';
+import { generateId } from '@labkey/components';
+
+import { getPlotConfigFromOptions } from "../utils";
+import { CurveFitData, PlotOptions } from "../models";
+import { PlotButtonBar } from "./PlotButtonBar";
+
+interface Props {
+    protocolId: number,
+    runId: number,
+    runPropertiesRow: {[key: string]: any},
+    data: any[],
+    plotOptions: PlotOptions,
+    curveFitData: CurveFitData,
+    columnInfo: {[key: string]: any}
+}
+
+interface State {
+    plotId: string
+}
+
+export class CalibrationCurvePanel extends PureComponent<Props, State> {
+    state: Readonly<State> = {
+        plotId: generateId('calibration-curve-')
+    };
+
+    componentDidMount(): void {
+        this.renderPlot();
+    }
+
+    componentDidUpdate(prevProps: Readonly<Props>, prevState: Readonly<State>, snapshot?: any): void {
+        this.renderPlot();
+    }
+
+    getPlotElement() {
+        return $('#' + this.state.plotId);
+    }
+
+    getPlotWidth(): number {
+        return $('.plot-panel-section').width();
+    }
+
+    renderPlot() {
+        const { runPropertiesRow, plotOptions, data, curveFitData, columnInfo } = this.props;
+        const LABKEY = getServerContext();
+
+        this.getPlotElement().html('<div class="loading-msg"><i class="fa fa-spinner fa-pulse"></i> Rendering plot...</div>');
+
+        setTimeout(() => {
```

Comment:
This setTimeout seems odd. Is it a re-render issue?

Reply:
It is just to help the plot refresh slow down a bit and look more natural so that it briefly show the user the "Rendering plot..." message. Without it, the plot just sort of flashes and is updated, so this looks better in my opinion.